polaris-service/src/test/java/org/apache/polaris/service/task/TableCleanupTaskHandlerTest.java

```diff
@@ -354,14 +395,207 @@ public void testTableCleanupMultipleSnapshots() throws IOException {
               taskEntity ->
                   assertThat(taskEntity)
                       .returns(PolarisEntityType.TASK.getCode(), PolarisBaseEntity::getTypeCode)
-                      .extracting(entity -> TaskEntity.of(entity))
+                      .extracting(TaskEntity::of)
                       .returns(
                           new ManifestFileCleanupTaskHandler.ManifestCleanupTask(
                               tableIdentifier,
                               Base64.encodeBase64String(ManifestFiles.encode(manifestFile3))),
                           entity ->
                               entity.readData(
-                                  ManifestFileCleanupTaskHandler.ManifestCleanupTask.class)));
+                                  ManifestFileCleanupTaskHandler.ManifestCleanupTask.class))
+          );
+
+      assertThat(StatsCleanupTasks)
+              .hasSize(2)
+              .satisfiesExactlyInAnyOrder(
+                      taskEntity ->
+                              assertThat(taskEntity)
+                                      .returns(PolarisEntityType.TASK.getCode(), PolarisBaseEntity::getTypeCode)
+                                      .extracting(TaskEntity::of)
+                                      .returns(AsyncTaskType.TABLE_CONTENT_CLEANUP, TaskEntity::getTaskType)
+                                      .returns(
+                                              new TableContentCleanupTaskHandler.TableContentCleanupTask(
+                                                      tableIdentifier, statisticsFile1.path()),
+                                              entity -> entity.readData(
+                                                      TableContentCleanupTaskHandler.TableContentCleanupTask.class)),
+                      taskEntity ->
+                              assertThat(taskEntity)
+                                      .returns(PolarisEntityType.TASK.getCode(), PolarisBaseEntity::getTypeCode)
+                                      .extracting(TaskEntity::of)
+                                      .returns(AsyncTaskType.TABLE_CONTENT_CLEANUP, TaskEntity::getTaskType)
+                                      .returns(
+                                              new TableContentCleanupTaskHandler.TableContentCleanupTask(
+                                                      tableIdentifier, statisticsFile2.path()),
+                                              entity -> entity.readData(
+                                                      TableContentCleanupTaskHandler.TableContentCleanupTask.class))
+              );
+    }
+  }
+
+  @Test
+  public void testTableCleanupMultipleMetadata() throws IOException {
```

Comment:
3. For the multiple metadata with multiple snapshot case, verify the ability to delete all of the statistics files and previous metadata